applications/crossbar/src/api_util.erl

```diff
@@ -868,17 +870,28 @@ ensure_content_type(CT) -> CT.
 %% (the final module in the chain) accepts this verb parameter pair.
 %% @end
 %%--------------------------------------------------------------------
--spec does_resource_exist(cb_context:context()) -> boolean().
--spec does_resource_exist(cb_context:context(), list()) -> boolean().
+
+-spec does_resource_exist(cb_context:context()) -> resource_exists().
+-spec does_resource_exist(cb_context:context(), list()) -> resource_exists().
 does_resource_exist(Context) ->
     does_resource_exist(Context, cb_context:req_nouns(Context)).
 
 does_resource_exist(Context, [{Mod, Params}|_]) ->
-    Event = ?MODULE:create_event_name(Context, <<"resource_exists.", Mod/binary>>),
-    Responses = crossbar_bindings:map(Event, Params),
-    crossbar_bindings:any(Responses) and 'true';
-does_resource_exist(_Context, _ReqNouns) ->
-    'false'.
+    Event = api_util:create_event_name(Context, <<"resource_exists.", Mod/binary>>),
+    Responses = case catch crossbar_bindings:map(Event, [Context | Params]) of
+                    {'EXIT', _Reason} -> [];
+                    Responses_  -> Responses_
+                end,
+    SuccessResponses = crossbar_bindings:succeeded(Responses),
+    case SuccessResponses of
+        [] -> {'false', hd([X || {'false', X} <- Responses])};
```

Comment:
Why not call a function to find the first {'false', Response} tuple and return it, rather than going through the whole list, deconstructing the tuple only to build it again?